cmd/variables.go

```diff
@@ -0,0 +1,408 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/satori/go.uuid"
+	"github.com/skuid/skuid-cli/platform"
+	"github.com/skuid/skuid-cli/text"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+type DataService struct {
+	ID          string    `json:"id"`
+	Name        string    `json:"name"`
+	Host        string    `json:"host"`
+	Port        string    `json:"port"`
+	Version     string    `json:"version"`
+	IsActive    bool      `json:"is_active"`
+	CreatedByID string    `json:"created_by_id"`
+	UpdatedByID string    `json:"updated_by_id"`
+	CreatedAt   time.Time `json:"created_at"`
+	UpdatedAt   time.Time `json:"updated_at"`
+}
+
+type EnvSpecificConfig struct {
+	ID              string `json:"id"`
+	OrganizationID  string `json:"organization_id"`
+	Name            string `json:"name"`
+	Value           string `json:"value"`
+	IsSecret        bool   `json:"is_secret"`
+	IsManaged       bool   `json:"is_managed"`
+	DataServiceID   string `json:"data_service_id"`
+	DataServiceName string
+	CreatedByID     string    `json:"created_by_id"`
+	UpdatedByID     string    `json:"updated_by_id"`
+	CreatedAt       time.Time `json:"created_at"`
+	UpdatedAt       time.Time `json:"updated_at"`
+}
+
+const fakeDefaultDataServiceId = "153b1f3e-e35a-4bb8-90a4-abbcc95fe15c"
+
+func isDefaultDs(ds string) bool {
+	defaultDS := map[string]interface{}{
+		"":                       nil,
+		fakeDefaultDataServiceId: nil,
+		"default":                nil,
+	}
+	_, ok := defaultDS[ds]
+	return ok
+}
+
+func getDataServices(api *platform.RestApi) (map[string]DataService, error) {
+	dspath := "/objects/dataservice"
+	dsStream, err := api.Connection.MakeRequest(
+		http.MethodGet,
+		dspath,
+		nil,
+		"application/json",
+	)
+	if err != nil {
+		return nil, errors.New("Error requesting Data Service list.")
+	}
+	var dataservices []DataService
+	if err = json.NewDecoder(*dsStream).Decode(&dataservices); err != nil {
+		return nil, errors.New("Could not parse Data Service list response.")
+	}
+	dsmap := make(map[string]DataService, len(dataservices))
+	for _, ds := range dataservices {
+		dsmap[ds.ID] = ds
+	}
+	return dsmap, nil
+}
+
+func findDataServiceId(api *platform.RestApi, name string) (string, error) {
+	if isDefaultDs(variabledataservice) {
+		return fakeDefaultDataServiceId, nil
+	}
+	if _, err := uuid.FromString(variabledataservice); err == nil {
+		return variabledataservice, nil
+	}
+	// Match the name with an existing Private Data Service
+	dataservices, err := getDataServices(api)
+	if err != nil {
+		return "", err
+	}
+	dataServiceId := ""
+	found := false
+	for _, ds := range dataservices {
+		if ds.Name == name {
+			found = true
+			dataServiceId = ds.ID
+			break
+		}
+	}
+	if !found {
```

Comment:
really minor but instead of using the `found` variable, could alternatively just return right here rather than using a break, and then remove the `if !found {` clause below and just return the error if we fail to find a ds within this loop